lib/gtfs/data.ex

```diff
@@ -274,43 +278,71 @@ defmodule Gtfs.Data do
     )
   end
 
-  @spec timepoint_ids_for_routes([RoutePattern.t()], MapSet.t(Route.id()), trips_by_id()) ::
-          timepoint_ids_by_route()
-  defp timepoint_ids_for_routes(route_patterns, route_ids, trips) do
+  @spec all_timepoints_by_id(binary()) :: Timepoint.timepoints_by_id()
+  defp all_timepoints_by_id(timepoints_data) do
+    timepoints_data
+    |> Csv.parse(parse: &Timepoint.from_csv_row/1)
+    |> Map.new(fn timepoint -> {timepoint.id, timepoint} end)
+  end
+
+  @spec timepoints_for_routes(
+          [RoutePattern.t()],
+          MapSet.t(Route.id()),
+          trips_by_id(),
+          Timepoint.timepoints_by_id()
+        ) ::
+          timepoints_by_route()
+  defp timepoints_for_routes(route_patterns, route_ids, trips, timepoints_by_id) do
     Enum.reduce(route_ids, %{}, fn route_id, acc ->
-      Map.put(acc, route_id, timepoint_ids_for_route(route_patterns, route_id, trips))
+      Map.put(
+        acc,
+        route_id,
+        timepoints_for_route(route_patterns, route_id, trips, timepoints_by_id)
+      )
```

Comment:
This already existed, so you don't have to change it, but if you wanted to, this could be changed to `Map.new(route_ids, fn route_id -> {route_id, timepoints_for_route(...)} end)`